### PR TITLE
Run test in parallel batches

### PR DESCRIPTION
**Pull request by jeanphilippeD**

### CODE REVIEW

**Comment by jeanphilippeD** [comment]
I was having a try at implementing this, most of this code is new to me
so i don't know if there are better way to achieve the same result.

Follow review suggestion to use chunks to run test in parallel.
Set default to 16 which works well even on my limited laptop,
and which should benefit better machine.

To run with a different batch size:
BINDGEN_TEST_BATCH_SIZE=32 cargo test

On my machine:
1 parallel test takes 3'53
2 parallel test takes 2'10
8 parallel test takes 2'08
32 parallel test takes 2'07

File: tests/tests.rs [hunk 1]
```diff
@@ -83,25 +83,35 @@ fn run_bindgen_tests() {
             Some(Some("hpp")) => true,
             _ => false,
         }
+    }).collect::<Vec<_>>();
+
+    let batch_size = env::var("BINDGEN_TEST_BATCH_SIZE")
+        .ok()
+        .map(|x| x.parse::<usize>().ok())
```
Comment (emilio) [inline tests/tests.rs]: nit: use `and_then` here and you'll save the following `unwrap_or(None)`.

File: tests/tests.rs [hunk 2]
```diff
@@ -83,25 +83,35 @@ fn run_bindgen_tests() {
...
+        .unwrap_or(None)
+        .unwrap_or(16);
```
Comment (emilio) [inline tests/tests.rs]: Could you move this number to a constant at the top of the file?

File: tests/tests.rs [hunk 3]
```diff
@@ -83,25 +83,35 @@ fn run_bindgen_tests() {
...
+        x.into_iter().filter_map(|(path, mut child)| {
             let passed = child.wait()
                 .expect("Should wait on child process")
                 .success();
 
             if passed { None } else { Some((path, child)) }
         })
-        .collect();
+    })
+    .flat_map(|x| x )
```
Comment (emilio) [inline tests/tests.rs]: use `flat_map` above (`children.flat_map(`), and you can remove this line.

**Comment by jeanphilippeD** [comment]
Thank you very much for your comments, they were very helpful.

**Pull request merged**